index.rst

```diff
@@ -183,8 +208,33 @@ shall be ignored.
 Suggestions for Plugin Developers
 =================================
 
-TODO. For now please read the `Plugin Guidelines`_ on GitHub wiki. 
+TODO. For now please read the `Plugin Guidelines`_ on GitHub wiki.
+
+Versioning
+==========
+
+*This section applies beginning with v0.13 of this specification.*
+
+This specification has a version, tagged in the `specification repository`_.
+Each specification version corresponds to the same version in the
+`core-tests repository`_.
+
+Each EditorConfig core, to pass the core tests, must process version
+numbers given with the ``-b`` switch, and must report version numbers when
+given ``-v`` or ``--version``.  The version numbers used for ``-b``, ``-v``,
+and ``--version`` are versions of this specification.  For example, the
+Vimscript core might respond to ``-v`` with:
+
+::
+
+  EditorConfig Vimscript core v1.0.0 - Specification Version 0.13
 
+Cores, plugins, or editors supporting EditorConfig have their own version
+numbers.  Those version numbers are independent of the version number of
+this specification.
```

Comment:
Do you think we can provide a helper regex here, so that programs can distinguish this uniform format from the old ones?

Reply:
We can certainly provide regexes, but I'm not sure I understand what you mean.  I think you are referring to the regex at https://github.com/editorconfig/editorconfig-core-test/blob/6c8fe6815b12f96f4b357d610ee1cd8da074880a/cli/CMakeLists.txt#L32 .  @florianb in https://github.com/editorconfig/editorconfig/issues/395#issuecomment-604300415 requested formatting the version numbers per semver - maybe we should discuss there or in a new issue.

Reply:
This is to help users who need to process the output of version numbers. What I meant is something like:

    if the output of $(editorconfig -v) matches the given regex; then
        process according to the new format
    else
        process according to the old format
    endif

Reply:
I think I may not understand.  What is the problem you would like to solve by including that logic in the specification?  I am happy to write text but do not want to do so unless I am sure it matches your expectation.

Each new version of the specification will change the way of processing somewhat.  Would a table of releases perhaps help?

Reply:
Sorry, I did not mean to write such a full logic. What I meant is to provide a regex so that it can be used to determine whether the version string matches the current format. In other words, it might be useful to have a regex that extracts the two most important version numbers in the string. Does this clarify?

Reply:
No worries, but I'm still struggling with the use case.  What if we put a link in the specification pointing directly to the relevant test?  Then, if we changed the tests, we wouldn't also have to change the specification.

Currently the tests require `vX.Y.Z` ([here](https://github.com/jjn1056/DBIx-Class-Migration/issues/130#issuecomment-565671773)).  We could put that regex into the spec if you wanted, but there's still @florianb's open https://github.com/editorconfig/editorconfig/issues/395#issuecomment-604300415 suggesting full semver v2.0.0 compliance.

Reply:
Thank you very much for that great work @cxw42!

When reading this proposal and the discussion the first thing which popped into my mind was that i'd like to ask for a machine readable representation as well. I scripted a lot of stuff for Infrastructure-Maintenance and it's often handy to easily compare or parse version numbers without having to code that much.

Reply:
sure - let's have a vote on your suggestion to semver the specification, then see what updates make sense.  If we do go with semver, we can point people to existing parse/compare libraries such as https://www.npmjs.com/package/semver .

Reply:
Vote requested at https://github.com/editorconfig/editorconfig/issues/395#issuecomment-628557400